amida-tech/amida-messaging-microservice
# Feature/ser 33 new branch

If this PR fixes a bug, you _must_ add test cases representative of the bug.
#### What's this PR do?
- Controller function for Count() written
- Made sure the special case of sending a message to self is taken care of
- Tests written
#### Related JIRA tickets:
https://jira.amida-tech.com/browse/SER-33
https://jira.amida-tech.com/browse/SER-122
#### How should this be manually tested?
- For counting all messages:
1. Send messages from 3 users to a single user `user0` using `http://localhost:4000/api/message/send`
2. `user0` also sends message to himself
3. `http://localhost:4000/api/message/count?option=all` should return only 4 messages
4. Make sure all of them have different `originalMessageId`s.
- For counting unread messages:
1. Send messages from 3 users to a single user `uer0` using `http://localhost:4000/api/message/send`
2. `user0` also sends message to himself
3. Read one of the messages using `http://localhost:4000/api/message/get/<id>`
4. `http://localhost:4000/api/message/count?option=all` should return only 3 messages
5. Make sure `readAt = null` for all of them

#### Any background context you want to provide?
- Sequelize is updated to 4.23.2
- Added `originalMessageId` as an attribute to be displayed in count() in controller.js
#### Screenshots (if appropriate):
